### ISSUE TITLE
Template usage discovery

### ISSUE THREAD

**Issue by drewfish** [screwdriver-cd/screwdriver#804]
As a template author, I would like to find a list of pipelines which are using my template. This is useful so that I can get an idea of which features are being used so I have a sense of if/whan/how to support or deprecate those features.

A UI would be nice, but also an API call would be helpful so that I can build tooling.

It's understandable that private pipelines wouldn't show up in this list.

**Comment by drewfish** [comment]
Oh hmm... might be nice to have a way to message template users too. Some ideas:
* send an email
* add a alert header to some part of the UI (pipeline page or something)

**Comment by d2lam** [comment]
That's a valid use case. However, another way I can think of this is - for example: do npm package owners care who uses their package? Is there a way for npm package owners to see who uses their package?

Just some questions we should think about. But I do agree with you that it would be pretty nice to have this.

**Comment by stjohnjohnson** [comment]
@d2lam Yes, they do: https://www.npmjs.com/browse/depended/mocha

**Comment by drewfish** [comment]
Yeah the communication channel can be tricky. As a template author I can setup a separate communication channel (google groups, tumblr blog, etc) and ask users to follow along. The "list pipelines using my template" is the heart of this request, since from that I can often answer a lot of questions I might have of the users of my template (from directly inspecting their pipeline definitions).

**Comment by d2lam** [comment]
Cool! I genuinely wanted to know 👍 
I wonder how they implemented that. Do they keep a database of who uses what? Like each time a package is published, it looks through its `package.json` and update the dependencies in the database❓
